### DISCUSSION TITLE
Rand error while building the examples

### DISCUSSION THREAD

**Discussion by adelmagteleco** [open-quantum-safe/liboqs#1737] [Q&A] [answered]
When I put this command: cmake --build liboqs-cpp/build --target examples --parallel 8
I get this error:
[ 33%] Building CXX object CMakeFiles/sig.dir/examples/sig.cpp.o
[ 33%] Building CXX object CMakeFiles/kem.dir/examples/kem.cpp.o
[ 50%] Building CXX object CMakeFiles/rand.dir/examples/rand.cpp.o
In file included from /home/magissss/Documentos/liboqs-cpp/examples/rand.cpp:8:
/home/magissss/Documentos/liboqs-cpp/include/rand/rand.hpp: In function ‘void oqs::rand::randombytes_nist_kat_init_256bit(const bytes&, const bytes&)’:
/home/magissss/Documentos/liboqs-cpp/include/rand/rand.hpp:92:12: error: ‘OQS_randombytes_nist_kat_init_256bit’ is not a member of ‘oqs::C’
92 | C::OQS_randombytes_nist_kat_init_256bit(entropy_input.data(),
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/magissss/Documentos/liboqs-cpp/include/rand/rand.hpp:96:8: error: ‘OQS_randombytes_nist_kat_init_256bit’ is not a member of ‘oqs::C’
96 | C::OQS_randombytes_nist_kat_init_256bit(entropy_input.data(), nullptr);
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/magissss/Documentos/liboqs-cpp/examples/rand.cpp: In function ‘int main()’:
/home/magissss/Documentos/liboqs-cpp/examples/rand.cpp:21:45: error: ‘OQS_RAND_alg_nist_kat’ was not declared in this scope; did you mean ‘OQS_RAND_alg_system’?
21 | oqs::rand::randombytes_switch_algorithm(OQS_RAND_alg_nist_kat);
| ^~~~~~~~~~~~~~~~~~~~~
| OQS_RAND_alg_system
gmake[3]: *** [CMakeFiles/rand.dir/build.make:76: CMakeFiles/rand.dir/examples/rand.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:209: CMakeFiles/rand.dir/all] Error 2
gmake[2]: *** Se espera a que terminen otras tareas....
[ 66%] Linking CXX executable sig
[ 83%] Linking CXX executable kem
[ 83%] Built target sig
[ 83%] Built target kem
gmake[1]: *** [CMakeFiles/Makefile2:164: CMakeFiles/examples.dir/rule] Error 2
gmake: *** [Makefile:179: examples] Error 2

It's my first time working with liboqs so I would erally appreciate any help, thank you so much!

**Comment by baentsch** [comment]
Thanks for this bug report. This most likely is an unintended downstream side-effect of https://github.com/open-quantum-safe/liboqs/pull/1667. Tagging @SWilson4 fyi. Issue already created in the suitable sub project [liboqs-cpp](https://github.com/open-quantum-safe/liboqs-cpp/issues/17). I don't quite understand how/why this issue did not get caught by CI: Tagging @vsoftco for insights on that.

**Reply by baentsch** [reply, marked as answer]
Resolved as per https://github.com/open-quantum-safe/liboqs-cpp/issues/17
